src/AutoSchedule.cpp

```diff
@@ -3436,9 +3426,32 @@ string generate_schedules(const vector<Function> &outputs, const Target &target,
     // used by some extern Funcs.
     debug(2) << "Determining all unbounded functions...\n";
     set<string> unbounded = get_unbounded_functions(pipeline_bounds, env);
+    if (!unbounded.empty()) {
+        // If some functions are unbounded, we should inline those directly.
+        // Also, we need to recompute 'env' and re-initialize 'costs' and
```

Comment:
Should we be just doing this first, rather than redoing all this analysis?

Reply:
I was just thinking that it's slightly messy to redo all the analysis, but looking closer at it, get_unbounded_functions depends on most of the analysis, so we can't really reorder it like I was hoping.

Reply:
Yeah, that's the annoying part. It needs the bounds computed in the previous steps before get_unbounded_functions is called. And since the inlining of unbounded funcs remove some the inlined funcs from the env, we need to re-compute the cost etc to match the new env.